catalog/dags/popularity/dag_factory.py

```diff
@@ -0,0 +1,181 @@
+        dag_id=dag_id,
+    )
+
+
+@task
+def get_providers_update_confs(
+    postgres_conn_id: str, popularity_refresh: PopularityRefresh
+):
+    """
+    Build a list of DagRun confs for each provider of this media type. The confs will
+    be used by the `batched_update` DAG to perform a batched update of all existing
+    records, to recalculate their standardized_popularity with the new popularity
+    constant. Providers that do not support popularity data are omitted.
+    """
+    # For the media type, get a list of the providers who support popularity data
+    providers = sql.get_providers_with_popularity_data_for_media_type(
+        postgres_conn_id, popularity_refresh.media_type
+    )
+
+    # Once the new popularity constants are calculated, they will automatically be used
+    # to calculate popularity scores at ingestion. The popularity refresh need only
+    # update the scores for records which were last updated before this time.
+    last_updated_time = datetime.datetime.utcnow()
+
+    # For each provider, create a conf that will be used by the batched_update to
+    # refresh standardized popularity scores.
+    return [
+        {
+            # Uniquely identify the query
+            "query_id": (
+                f"{provider}_popularity_refresh_{last_updated_time.strftime('%Y%m%d')}"
+            ),
+            "table_name": popularity_refresh.media_type,
+            # Query used to select records that should be refreshed
+            "select_query": (
+                f"WHERE provider='{provider}' AND updated_on <"
+                f"'{last_updated_time.strftime('%Y-%m-%d %H:%M:%S')}'"
+            ),
+            # Query used to update the standardized_popularity
+            "update_query": sql.format_update_standardized_popularity_query(
+                popularity_refresh.media_type
+            ),
+            "batch_size": 10_000,
+            "update_timeout": (
+                popularity_refresh.refresh_popularity_batch_timeout.total_seconds()
+            ),
+            "dry_run": False,
+            "resume_update": False,
+        }
+        for provider in providers
+    ]
+
+
+def create_popularity_refresh_dag(popularity_refresh: PopularityRefresh):
+    """
+    Instantiate a DAG for a popularity refresh.
+
+    For the given media type, this DAG will recalculate the popularity constants and
+    then refresh all existing standardized popularity scores with the new constant.
+
+    Required Arguments:
+
+    popularity_refresh: dataclass containing configuration information for the DAG
+    """
+    default_args = {
+        **DAG_DEFAULT_ARGS,
+        **popularity_refresh.default_args,
+    }
+
+    dag = DAG(
+        dag_id=popularity_refresh.dag_id,
+        default_args=default_args,
+        start_date=popularity_refresh.start_date,
+        schedule=popularity_refresh.schedule,
+        max_active_runs=1,
+        catchup=False,
+        doc_md=__doc__,
+        tags=["popularity_refresh"],
```

Comment:
Do you think it would make sense for this to also have the `"data_refresh"` tag? Or should we leave it off because it's independent?

Reply:
When the project is complete, the data and popularity refreshes should be totally decoupled. There's definitely a case to be made that they're _related_ but I think it might be more confusing to use the `data_refresh` tag here, in case it implies a relationship that no longer exists. But that's not a strongly held opinion :)